src/modules/modem_module.c

```diff
@@ -612,29 +613,52 @@ static void on_all_states(struct modem_msg_data *msg)
 	}
 }
 
-static void message_handler(struct modem_msg_data *msg)
+static void modem_module_thread_fn(void)
 {
-	switch (state) {
-	case STATE_DISCONNECTED:
-		on_state_disconnected(msg);
-		break;
-	case STATE_CONNECTING:
-		on_state_connecting(msg);
-		break;
-	case STATE_CONNECTED:
-		on_state_connected(msg);
-		break;
-	case STATE_SHUTTING_DOWN:
-		LOG_WRN("No action allowed in STATE_SHUTTING_DOWN");
-		break;
-	default:
-		LOG_WRN("Invalid state: %d", state);
-		break;
+	int err;
+	struct modem_msg_data msg;
+
+	self.thread_id = k_current_get();
+
+	module_start(&self);
```

Comment:
A little out of scope, but this function should be called ```setup()``` as in the other modules.

Reply:
Changed `modem_setup` to `setup`